storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -28,94 +32,270 @@
+            while (true) {
+                SchedulingResult result = null;
+                try {
+                    //Need to re prepare scheduling strategy with cluster and topologies in case scheduling state was restored
+                    rasStrategy.prepare(this.topologies, this.cluster, this.userMap, this.nodes);
+                    result = rasStrategy.schedule(td);
+                } catch (Exception e) {
+                    LOG.error("Exception thrown when running strategy {} to schedule topology {}. Topology will not be scheduled! Error: {} StackTrace: {}"
+                            , rasStrategy.getClass().getName(), td.getName(), e.getMessage(), Arrays.toString(e.getStackTrace()));
+                    this.restoreCheckpointSchedulingState(schedulingState);
+                    //since state is restored need the update User topologySubmitter to the new User object in userMap
+                    topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+                    topologySubmitter.moveTopoFromPendingToInvalid(td);
+                    this.cluster.setStatus(td.getId(), "Unsuccessful in scheduling - Exception thrown when running strategy {}"
+                            + rasStrategy.getClass().getName() + ". Please check logs for details");
+                }
+                LOG.debug("scheduling result: {}", result);
+                if (result != null && result.isValid()) {
+                    if (result.isSuccess()) {
+                        try {
+                            if (mkAssignment(td, result.getSchedulingResultMap())) {
+                                topologySubmitter.moveTopoFromPendingToRunning(td);
+                                this.cluster.setStatus(td.getId(), "Running - " + result.getMessage());
+                            } else {
+                                this.restoreCheckpointSchedulingState(schedulingState);
+                                //since state is restored need the update User topologySubmitter to the new User object in userMap
+                                topologySubmitter = this.userMap.get(td.getTopologySubmitter());
+                                topologySubmitter.moveTopoFromPendingToAttempted(td);
+                                this.cluster.setStatus(td.getId(), "Unsuccessful in scheduling - Unable to assign executors to nodes. Please check logs for details");
                             }
-                            assignedMemOnHeap += targetSlot.getAllocatedMemOnHeap();
-                            assignedMemOffHeap += targetSlot.getAllocatedMemOffHeap();
-                            assignedCpu += targetSlot.getAllocatedCpu();
+                        } catch (IllegalStateException ex) {
+                            LOG.error(ex.toString());
+                            LOG.error("Unsuccessful in scheduling - IllegalStateException thrown when attempting to assign executors to nodes. Error: {} StackTrace: {}",
```

Comment:
`LOG.error("Unsuccessful in scheduling - IllegalStateException thrown when attempting to assign executors to nodes.", ex);`